File: code-generator/src/main/java/org/xi/quick/codegenerator/service/TableService.java
package org.xi.quick.codegenerator.service;

import org.xi.quick.codegenerator.model.TableModel;

import java.util.List;
import java.util.Set;

public interface TableService {

    /**
     * 获取所有表名
     *
     * @return
     */
    Set<String> getAllTableNameList();

    /**
     * 获取单表
     *
     * @param tableName
     * @return
     */
    TableModel getTable(String tableName);

    /**
     * 获取所有列表
     *
     * @return
     */
    List<TableModel> getAllTables();
}

File: code-generator-kt/src/main/kotlin/org/xi/quick/codegeneratorkt/service/TableService.kt
package org.xi.quick.codegeneratorkt.service

import org.xi.quick.codegeneratorkt.model.TableModel

interface TableService {

    /**
     * 获取所有表名
     *
     * @return
     */
    fun getAllTableNameList(): Set<String>

    /**
     * 获取单表
     *
     * @param tableName
     * @return
     */
    fun getTable(tableName: String): TableModel?

    /**
     * 获取所有列表
     *
     * @return
     */
    fun getAllTables(): List<TableModel>
}